packages/tools/src/utilities/segmentation/thresholdVolumeByRange.ts

```diff
@@ -37,15 +48,161 @@ function thresholdVolumeByRange(
+    const upper = thresholdVolumeInformation[i].upper;
+
+    volumeInfoList.push({
+      imageData,
+      referenceValues,
+      lower,
+      upper,
+      spacing,
+      dimensions,
+      volumeSize,
+    });
+  }
+
+  // global variables used in calbackOverlap function
+  let overlaps, total, range;
+
+  /**
+   * As we are working with volumes of different dimensions, there will be no
+   * 1 to 1 mapping. So we need to work with the idea of voxel overlaps
+   * (1 to many mappings). We consider all intersections valid, to avoid the
+   * complexity to calculate a minimum voxel intersection percentage. This
+   * function, given a voxel center and spacing, calculates the overlap of the
+   * voxel with another volume and range check the voxels in the overlap. Three
+   * situations can occur: all voxels pass the range check, some voxels pass or
+   * none voxels pass. The overlapType parameter indicates if the user requires
+   * all voxels pass (overlapType = 1) or any voxel pass (overlapType = 0)
+   */
+  const testOverlapRange = (volumeInfo, voxelSpacing, voxelCenter) => {
+    const { imageData, dimensions, lower, upper } = volumeInfo;
+
+    const overlapBounds = getVoxelOverlap(
+      imageData,
+      dimensions,
+      voxelSpacing,
+      voxelCenter
+    );
+
+    // reset global variables and setting the range check
+    total = 0;
+    overlaps = 0;
+    range = { lower, upper };
+
+    let overlapTest = false;
+
+    // check all voxel overlaps
+    pointInShapeCallback(imageData, () => true, callbackOverlap, overlapBounds);
+
+    if (overlapType === 0) {
+      overlapTest = overlaps > 0; // any voxel overlap is accepted
+    } else if (overlapType == 1) {
+      overlapTest = overlaps === total; // require all voxel overlaps
+    }
+    return overlapTest;
+  };
+
+  // range checks a voxel in a volume with same dimension as the segmentation
+  const testRange = (volumeInfo, pointIJK) => {
+    const { imageData, referenceValues, lower, upper } = volumeInfo;
     const offset = imageData.computeOffsetIndex(pointIJK);
+
     const value = referenceValues[offset];
     if (value <= lower || value >= upper) {
-      return;
+      return false;
+    } else {
+      return true;
+    }
+  };
+
+  /**
+   * Given the center of a voxel in world coordinates, calculate the voxel
+   * corners in world coords to calculate the voxel overlap in another volume
+   */
+  const getVoxelOverlap = (
+    imageData,
+    dimensions,
+    voxelSpacing,
+    voxelCenter
+  ) => {
+    const voxelCornersWorld = [];
+    for (let i = 0; i < 2; i++) {
+      for (let j = 0; j < 2; j++) {
+        for (let k = 0; k < 2; k++) {
+          const point = voxelCenter;
+          point[0] = point[0] + ((i * 2 - 1) * voxelSpacing[0]) / 2;
+          point[1] = point[1] + ((j * 2 - 1) * voxelSpacing[1]) / 2;
+          point[2] = point[2] + ((k * 2 - 1) * voxelSpacing[2]) / 2;
+          voxelCornersWorld.push(point);
+        }
+      }
+    }
```

Comment:
if the reference (blue) is finer resolution

![image](https://user-images.githubusercontent.com/7490180/210886428-70dba594-f336-43cc-8e0e-e3b256257f68.png)

Then going half reference spacing -+ from the center of the grey one doesn't help

Reply:
But i guess the logic is true the other way around, what do you think

Reply:
I think the logic is correct. If the blue is the reference it should only affect the voxel it is inside.

![image](https://user-images.githubusercontent.com/114958722/212029133-3bcc0e9f-2148-4334-9c85-d66d1f18d88d.png)

And if it is in the corner of four voxels. It should affect them also

![image](https://user-images.githubusercontent.com/114958722/212029414-9ffd1f57-8519-43fd-89cf-c72d4eb03789.png)

Reply:
I don't think you are right, please double check the two scenarios again. 

The bounds should always be computed wrt the larger spacing, otherwise the overlaps are computed wrongly